### PR TITLE
[dotnet] fix build warnings

### PR DESCRIPTION
**Pull request by lucaspimentel**
## Motivation

In .NET project `utils/build/docker/dotnet`:
- fix errors that prevent building locally (outside of docker)
- fix (or suppress) build warnings

## Changes

- exclude directory `parametric/` from build
- fix target framework
- add missing null checks and nullability annotations
- remove some unused declarations
- fix some instances of string interpolation
- other miscellaneous fixes: typos, code formatting, etc
- suppress warnings for known issues (`NU1902`, `NU1903`, and `CA1416`)

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: utils/build/docker/dotnet/app.csproj [hunk 1]
```diff
@@ -1,32 +1,48 @@
...
+        <TargetFramework>net6.0</TargetFramework>
+        <RootNamespace>weblog</RootNamespace>
+        <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
+        <GenerateTargetFrameworkAttribute>false</GenerateTargetFrameworkAttribute>
+        <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+        <Nullable>enable</Nullable>
+
+        <!--
+          Ignore the following warnings:
+            NU1902: Package 'System.Data.SqlClient' 4.6.1 has a known moderate severity vulnerability, https://github.com/advisories/GHSA-8g2p-5pqh-5jmc
```
Comment (lucaspimentel) [inline utils/build/docker/dotnet/app.csproj:12]: @daniel-romano-DD: Is the use of `System.Data.SqlClient` 4.6.1 intentional or can we upgrade to a newer version without the vulnerabilities? See #1725.

File: utils/build/docker/dotnet/app.csproj [hunk 2]
```diff
@@ -1,32 +1,48 @@
...
+            NU1903: Package 'System.Data.SqlClient' 4.6.1 has a known high severity vulnerability, https://github.com/advisories/GHSA-98g6-xh36-x2p7
+            CA1416: This call site is reachable on all platforms. 'DirectoryEntry' is only supported on: 'windows'. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416)
```
Comment (lucaspimentel) [inline utils/build/docker/dotnet/app.csproj:14]: @NachoEchevarria: Do we run these tests on Windows? See #1765.

**Review by zacharycmontoya — APPROVED** [review]
I see that there are some open questions about upgrading the `System.Data.SqlClient` package, but regardless of that change this looks good to me as an other .NET dev 👍🏼

File: utils/build/docker/dotnet/app.csproj (hunk 2 above)
Comment (NachoEchevarria) [reply utils/build/docker/dotnet/app.csproj:14]: I think that these tests are only run on Linux. Still, this test is able to probe that the LDAP instrumentation works. I think that the only way to pass this feature without getting a warning would be disabling this warning for that particular statement.

**Comment by cbeauchesne** [comment]
Can you update your branch, @robertomonteromiguel just fixed the issue on debugger/replay mode

File: utils/build/docker/dotnet/app.csproj (hunk 2 above)
Comment (lucaspimentel) [reply utils/build/docker/dotnet/app.csproj:14]: Thanks for confirming!

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**